cli/src/lib.rs

```diff
@@ -1921,24 +1952,47 @@ fn idl_set_buffer(cfg_override: &ConfigOverride, program_id: Pubkey, buffer: Pub
             }
         };
 
-        // Build the transaction.
-        let latest_hash = client.get_latest_blockhash()?;
-        let tx = Transaction::new_signed_with_payer(
-            &[set_buffer_ix],
-            Some(&keypair.pubkey()),
-            &[&keypair],
-            latest_hash,
-        );
+        // print only mdde
+        if authority.is_some() {
+            let transaction = TransactionInstruction {
```

Comment:
Can we name this `instruction` rather than a `transaction` since it's an instruction?